src/main/java/seedu/pettracker/storage/Storage.java

```diff
@@ -181,40 +190,72 @@ private void parsePetFile(ArrayList<String> data) {
                 System.out.println("Pet name in file is empty");
             } catch (DuplicatePetException e) {
                 System.out.println("File contains duplicate pet names");
+            } catch (InvalidSeparatorException e) {
+                System.out.println("Output File contains invalid separator/invalid number of separator");
             }
         }
     }
 
     private void parseTaskFile(ArrayList<String> data) {
         for (String line : data) {
             try {
-                String taskName = getTaskName(line);
+                validateTaskDataSep(line);
                 LocalDate deadline = getDeadline(line);
-                TaskList.addTask(taskName, deadline);
-
+                String taskName = getTaskName(line);
                 String taskStatus = getTaskStatus(line);
-                if (taskStatus.startsWith("1")) {
+
+                TaskList.addTask(taskName, deadline);
+                if (taskStatus.equals("1")) {
                     int taskNumber = TaskList.getNumberOfTasks();
                     TaskList.markTask(taskNumber, true);
                 }
+            } catch (InvalidSeparatorException e) {
+                System.out.println("Task output file contains invalid separator/invalid number of separator. " +
```

Comment:
Additionally, since Ui is parsed into this method and the methods below, use Ui to print the statements to improve abstraction